server/mvcc/buckets/lease.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buckets
+
+import (
+	"encoding/binary"
+	"math"
+
+	"go.etcd.io/etcd/server/v3/lease/leasepb"
+	"go.etcd.io/etcd/server/v3/mvcc/backend"
+)
+
+func UnsafeCreateLeaseBucket(tx backend.BatchTx) {
+	tx.UnsafeCreateBucket(Lease)
+}
+
+func MustUnsafeGetAllLeases(tx backend.ReadTx) []*leasepb.Lease {
+	_, vs := tx.UnsafeRange(Lease, int64ToBytes(0), int64ToBytes(math.MaxInt64), 0)
+	ls := make([]*leasepb.Lease, 0, len(vs))
+	for i := range vs {
+		var lpb leasepb.Lease
+		err := lpb.Unmarshal(vs[i])
+		if err != nil {
+			panic("failed to unmarshal lease proto item")
+		}
+		ls = append(ls, &lpb)
+	}
+	return ls
+}
+
+func MustUnsafePutLease(tx backend.BatchTx, lpb *leasepb.Lease) {
+	key := int64ToBytes(lpb.ID)
+
+	val, err := lpb.Marshal()
+	if err != nil {
+		panic("failed to marshal lease proto item")
+	}
+	tx.UnsafePut(Lease, key, val)
+}
+
+func UnsafeDeleteLease(tx backend.BatchTx, lpb *leasepb.Lease) {
+	tx.UnsafeDelete(Lease, int64ToBytes(lpb.ID))
+}
+
+func MustUnsafeGetLease(tx backend.BatchTx, lpb *leasepb.Lease) *leasepb.Lease {
```

Comment:
If its not crucial for performance (I assume so), pls make the method take only LeaseId and return the lpb. 

Alternatively (not recommended): Please add a comment that it mutates `lpb`.

Reply:
Using leasepb here is not great, but I proposed it as a temporary solution. Using LeaseId here is not possible as it creates import cycle (lease module defines LeaseID, but also imports MustUnsafeGetLease from buckets, buckets need to import LeaseID). I will try a not recommended option of adding comment, and prepare a separate PR with proper solution.

The proper solution would be to have buckets (should be renamed to schema at that point) define a struct that consumes LeaseID. In lease module we will define an interface LeaseStorage based on the struct methods. As this struct will be injected to interface, lease module will no longer import buckets directly.

Reply:
`int64 leaseId` is also a mid-ground option.

Reply:
Done.